clesto/__init__.py

```diff
@@ -806,9 +806,201 @@ def __init__(self, data=None, torsion=None):
         # initialize element
         super(Surjection_element, self).__init__(data=data, torsion=torsion)
 
+    def pcompose(self, other, k):
+        '''partial composition'''
+
+        class Piece:
+            def __init__(self, piece, is_in_surj1, index_in_surj):
+                self.piece = piece
+                self.is_in_surj1 = is_in_surj1
+                self.index_in_surj = index_in_surj
+
+            def __getitem__(self, index):
+                return self.piece[index]
+
+            def __len__(self):
+                return len(self.piece)
+
+            def __eq__(self, other):
+                if isinstance(other, Piece):
+                    result = self.piece == other.piece
+                    result &= self.is_in_surj1 == other.is_in_surj1
+                    result &= self.index_in_surj == other.index_in_surj
+                    return result
+                return False
+
+        def table_arrangement(surj, only_dict=False):
+            '''Returns the table arrangement of a surjection, as a tuple.
```

Comment:
I thought table_arrangement was going to be a method in the class? I think that that might be a good idea since I suspect that this arrangement plays a role in the chain homotopy inverse TC: Surjection --> Barratt-Eccles defined in https://arxiv.org/abs/math/0204326 , but then again, maybe not, and I have never read this paper (it's in french).